### PR TITLE
Add support for missing services

### PR DESCRIPTION
**Pull request by francescofuggitti**
In this PR, we add support for the following missing services:

- Baremetal (ironic):
    - allocations
    - drivers
    - nodes
    - ports
- Blockstorage (cinder):
    - availability_zones
    - limits
    - qos
    - quotasets
    - quotasets_usage
    - snapshots
    - services
    - volume_backups
- Compute (nova):
    - serverusage
- Identity (keystone):
    - domains
    - domain_groups
    - projects
    - regions
    - registered limits
    - roles
    - services
    - users
    - users_keypairs
- Image (glance):
    - image_members
- Configure connection with Apache Superset

Also, we apply the following fixes:

- add spec validation
- apply formatting
- fix readme badges

Finally, this PR aims to update the following packages:

- bump net from v0.10.0 to v0.17.0 (PR #1) 
- bump gprc from v1.57.0 to v1.57.1 (PR #2)
- bump apache arrow from v13.0.0 to v15.0.0 (this depends on the approval and merge of the PRs (https://github.com/dihedron/cq-plugin-utils/pull/1 and https://github.com/dihedron/cq-plugin-utils/pull/2 and https://github.com/dihedron/cq-plugin-utils/pull/3)

### CODE REVIEW

File: resources/plugin/client.go [hunk 1]
```diff
@@ -11,7 +11,11 @@ import (
 	"github.com/cloudquery/plugin-sdk/v4/schema"
 	"github.com/cloudquery/plugin-sdk/v4/transformers"
 	"github.com/dihedron/cq-source-openstack/client"
-	"github.com/dihedron/cq-source-openstack/resources"
+	"github.com/dihedron/cq-source-openstack/resources/services/blockstorage"
+	"github.com/dihedron/cq-source-openstack/resources/services/compute"
+	"github.com/dihedron/cq-source-openstack/resources/services/identity"
+	"github.com/dihedron/cq-source-openstack/resources/services/image_service"
```
Comment (dihedron) [inline resources/plugin/client.go]: per uniformità con gli altri, chiamerei il package image e basta, non image_service.

**Comment by dihedron** [comment]
Non vedo le modifiche relative a:
1. supporto per include ed exclude filter.
2. supporto nel makefile per la sovrascrittura dei simboli pubblic (plugin.Version) attraverso linkr flag.

File: resources/internal/pattern_matcher/pattern_matcher.go [hunk 2]
```diff
@@ -51,29 +50,27 @@ func (pm *PatternMatcher) Match(value string) bool {
...
-	for _, in := range pm.include {
-		g = glob.MustCompile(in)
-		is_included = g.Match(value)
-		if is_included {
-			break
-		}
-	}
+    // Check if the value matches any pattern in the pm.include list
+    for _, pattern := range pm.include {
+		g = glob.MustCompile(pattern)
```
Comment (dihedron) [inline resources/internal/pattern_matcher/pattern_matcher.go]: Mustcompile causa panic; non può essere messo un panic su uno user input.

File: resources/internal/pattern_matcher/pattern_matcher.go [hunk 3]
```diff
@@ -51,29 +50,27 @@ func (pm *PatternMatcher) Match(value string) bool {
...
-		}
-	}
-
-	if is_included && !is_excluded {
-		return true
-	}
-	return false
+	// Check if the value matches any pattern in the excluded list
+    for _, pattern := range pm.exclude {
+		g = glob.MustCompile(pattern)
```
Comment (dihedron) [inline resources/internal/pattern_matcher/pattern_matcher.go]: MustCompile causa panic.

File: resources/internal/pattern_matcher/pattern_matcher.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	}
+}
+
+func WithExclude(list []string) Option {
+	return func(c *PatternMatcher) {
+		c.exclude = list
+	}
+}
+
+func WithIncludeFilter(value string) Option {
```
Comment (dihedron) [inline resources/internal/pattern_matcher/pattern_matcher.go]: Remove

File: resources/internal/pattern_matcher/pattern_matcher.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	return func(c *PatternMatcher) {
+		c.include = append(c.include, value)
+	}
+}
+
+func WithExcludeFilter(value string) Option {
```
Comment (dihedron) [inline resources/internal/pattern_matcher/pattern_matcher.go]: Remove

File: plugin/plugin.go [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package plugin
+
+const (
+	Name = "dihedron-openstack"
+	Kind = "source"
+)
+
+var Version = "development"
```
Comment (dihedron) [inline plugin/plugin.go]: check if constant can be overridden by LDFLAGS.

File: go.mod [hunk 7]
```diff
@@ -1,68 +1,129 @@
 module github.com/dihedron/cq-source-openstack
 
-go 1.21
-
-toolchain go1.21.0
+go 1.21.1
```
Comment (dihedron) [inline go.mod]: Bump to 1.22

File: go.mod [hunk 8]
```diff
@@ -1,68 +1,129 @@
...
 
 require (
-	github.com/apache/arrow/go/v13 v13.0.0-20230731205701-112f94971882
-	github.com/cloudquery/plugin-sdk/v4 v4.5.0
-	github.com/dihedron/cq-plugin-utils v0.0.0-20230816150611-90ab3d855604
+	github.com/apache/arrow/go/v15 v15.0.0
+	github.com/cloudquery/plugin-sdk/v4 v4.29.1
```
Comment (dihedron) [inline go.mod]: Bump to 4.30

File: client/client.go [hunk 9]
```diff
@@ -169,49 +178,59 @@ type serviceConfig struct {
 }
 
 var serviceConfigMap = map[ServiceType]serviceConfig{
-	IdentityV3: {
+	IronicV1: {
```
Comment (dihedron) [inline client/client.go]: Revert to BareMetal (see others too)

File: client/client.go [hunk 10]
```diff
@@ -142,25 +148,28 @@ func (c *Client) initServiceClient(key ServiceType) (*gophercloud.ServiceClient,
 
 const (
 	// defaults currently referring to Train
-	DefaultComputeV2Microversion      = "2.79"
-	DefaultIdentityV3Microversion     = "3.13"
-	DefaultBlockStorageV3Microversion = "3.59"
-	DefaultImageServiceV2Microversion = "2.9"
+	DefaultIronicV1Microversion   = "1.58"
```
Comment (dihedron) [inline client/client.go]: Revert to BareMetal, Compute....

File: client/spec.go [hunk 11]
```diff
@@ -1,28 +1,49 @@
...
+	if err != nil {
+		log.Error().Err(err).Msg("invalid endpoint URL")
+		return err
+	}
+	// Check that the endpoint URL is reachable
+	_, err = http.Get(*s.EndpointUrl)
+	if err != nil {
+		log.Error().Err(err).Msg("endpoint URL is unreachable")
+		return err
+	}
```
Comment (dihedron) [inline client/spec.go:102]: Add further tests on spec, e.g. check on fieelds when using AuthCredentials (see client.go for details)

File: resources/internal/utils/data_structs.go [hunk 12]
```diff
@@ -8,3 +8,7 @@ type Pair[K comparable, V any] struct {
 	Key   K `cq-name:"key"`
 	Value V `cq-name:"value"`
 }
+
+type Tag struct {
```
Comment (dihedron) [inline resources/internal/utils/data_structs.go]: Is this Tag struct used anywhere?

File: resources/plugin/client.go [hunk 13]
```diff
@@ -26,26 +33,29 @@ type Client struct {
 }
 
 func Configure(ctx context.Context, logger zerolog.Logger, spec []byte, opts plugin.NewClientOptions) (plugin.Client, error) {
+	config := &client.Spec{}
+	if err := json.Unmarshal(spec, config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal spec: %w", err)
+	}
+
```
Comment (dihedron) [inline resources/plugin/client.go:43]: tables := getTables(config)

File: resources/plugin/client.go [hunk 14]
```diff
@@ -26,26 +33,29 @@ type Client struct {
...
 	if opts.NoConnection {
 		return &Client{
 			logger: logger,
-			tables: getTables(),
+			tables: getTables(config),
```
Comment (dihedron) [inline resources/plugin/client.go]: tables : tables,

File: resources/plugin/client.go [hunk 15]
```diff
@@ -26,26 +33,29 @@ type Client struct {
...
-	if err := json.Unmarshal(spec, config); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal spec: %w", err)
-	}
-
 	syncClient, err := client.New(ctx, logger, config)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create client: %w", err)
 	}
+
+	tables := getTables(config)
```
Comment (dihedron) [inline resources/plugin/client.go]: Remove

File: resources/plugin/client.go [hunk 16]
```diff
@@ -74,23 +84,71 @@ func (*Client) Close(_ context.Context) error {
...
-		resources.Networks(),
-		resources.Ports(),
-		resources.Projects(),
-		resources.SecurityGroups(),
-		resources.SecurityGroupRules(),
-		resources.Users(),
-		resources.Volumes(),
+func getTables(spec *client.Spec) schema.Tables {
+	available_tables := schema.Tables{
+		ironic.Allocations(),
```
Comment (dihedron) [inline resources/plugin/client.go]: Rename packages to baremetal...

File: resources/plugin/client.go [hunk 17]
```diff
@@ -74,23 +84,71 @@ func (*Client) Close(_ context.Context) error {
...
+		keystone.Services(),
+		glance.Images(),
+		neutron.Networks(),
+		neutron.Ports(),
+		neutron.SecurityGroups(),
+		neutron.SecurityGroupRules(),
 	}
+
+	// must compile these patterns to be included
+	includesDaSpec := spec.IncludedTables
```
Comment (dihedron) [inline resources/plugin/client.go]: includesFromSpec

File: resources/plugin/client.go [hunk 18]
```diff
@@ -74,23 +84,71 @@ func (*Client) Close(_ context.Context) error {
...
+	// must compile these patterns to be excluded
+	excludesDaSpec := spec.ExcludedTables
```
Comment (dihedron) [inline resources/plugin/client.go]: excludesFromSpec

File: resources/plugin/client.go [hunk 19]
```diff
@@ -74,23 +84,71 @@ func (*Client) Close(_ context.Context) error {
...
+
+	// if includesDaSpec is empty, include everything
+	var pm *pattern_matcher.PatternMatcher
```
Comment (dihedron) [inline resources/plugin/client.go]:
```golang

pm := pattern_matcher.New(
    WithInclude(spec.includesFromSpec), // <--- check if input is valid before overwriting
    WithExclude(spec.includesFromSpec), // <--- check if input is valid before overwriting
)
```

File: resources/internal/pattern_matcher/pattern_matcher.go [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+
+	for _, option := range options {
+		option(pm)
+	}
+	return pm
+}
+
+func WithInclude(list []string) Option {
+	return func(c *PatternMatcher) {
+		c.include = list
```
Comment (dihedron) [inline resources/internal/pattern_matcher/pattern_matcher.go]:
```golang
if len(list) > 0 {
    c.include = list
}
```

File: resources/plugin/client.go [hunk 21]
```diff
@@ -74,23 +84,71 @@ func (*Client) Close(_ context.Context) error {
...
+			pattern_matcher.WithInclude(includesDaSpec),
+		)
+	} else {
+		pm = pattern_matcher.New(
+			pattern_matcher.WithInclude(includesDaSpec),
+			pattern_matcher.WithExclude(excludesDaSpec),
+		)
+	}
+
+	tables := make(schema.Tables, 0, len(available_tables))
```
Comment (dihedron) [inline resources/plugin/client.go]:
```golang
tables := []schemaTables{}
```

File: resources/services/cinder/attachments.go [hunk 22]
```diff
@@ -15,7 +17,7 @@ import (
 
 func Attachments() *schema.Table {
 	return &schema.Table{
-		Name:     "openstack_attachments",
+		Name:     "openstack_cinder_attachments",
```
Comment (dihedron) [inline resources/services/cinder/attachments.go]: openstack_compute_*

File: resources/services/cinder/qos.go [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		api.Logger().Err(err).Msg("error extracting QoS")
+		return err
+	}
+	for _, qos := range allQoS {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		qos := qos
```
Comment (dihedron) [inline resources/services/cinder/qos.go]: If bumped to 1.22, this becomes redundant.

File: resources/services/cinder/quotasets.go [hunk 24]
```diff
@@ -0,0 +1,73 @@
...
+	}
+	api.Logger().Debug().Int("count", len(allProjects)).Msg("projects retrieved")
+
+	projectIDs := []string{}
+	for _, project := range allProjects {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		project := project
```
Comment (dihedron) [inline resources/services/cinder/quotasets.go]: Redundant with Golang 1.22

File: resources/services/cinder/quotasets_usage.go [hunk 25]
```diff
@@ -0,0 +1,292 @@
...
+	}
+	api.Logger().Debug().Int("count", len(allProjects)).Msg("projects retrieved")
+
+	projectIDs := []string{}
+	for _, project := range allProjects {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		project := project
```
Comment (dihedron) [inline resources/services/cinder/quotasets_usage.go]: Rednundant with Golang 1.22

File: resources/services/cinder/quotasets_usage.go [hunk 26]
```diff
@@ -0,0 +1,292 @@
...
+		if err != nil {
+			api.Logger().Error().Err(err).Msg("error extracting quota sets for project " + projectID)
+			return err
+		}
+		res <- quotausageset
+	}
+	return nil
+}
+
+type QuotaUsageSet struct {
```
Comment (dihedron) [inline resources/services/cinder/quotasets_usage.go]: Maybe this whole struct is unnecessary???

File: resources/services/cinder/services.go [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+		return err
+	}
+	api.Logger().Debug().Int("count", len(allServices)).Msg("services retrieved")
+
+	for _, service := range allServices {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		service := service
```
Comment (dihedron) [inline resources/services/cinder/services.go]: Rednundant with Golang 1.22

File: resources/services/cinder/snapshots.go [hunk 28]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		api.Logger().Err(err).Msg("error extracting snapshots")
+		return err
+	}
+	for _, snapshot := range allSnapshots {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		snapshot := snapshot
```
Comment (dihedron) [inline resources/services/cinder/snapshots.go]: Rednundant with Golang 1.22

File: resources/services/cinder/volumes.go [hunk 29]
```diff
@@ -14,7 +16,7 @@ import (
 
 func Volumes() *schema.Table {
 	return &schema.Table{
-		Name:     "openstack_volumes",
+		Name:     "openstack_cinder_volumes",
```
Comment (dihedron) [inline resources/services/cinder/volumes.go]: cinder -> storage

File: resources/services/glance/image_members.go [hunk 30]
```diff
@@ -0,0 +1,73 @@
...
+		return err
+	}
+	api.Logger().Debug().Int("count", len(allMembers)).Msg("image members retrieved")
+
+	for _, member := range allMembers {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		member := member
```
Comment (dihedron) [inline resources/services/glance/image_members.go]: Rednundant with Golang 1.22

File: resources/services/glance/images.go [hunk 31]
```diff
@@ -24,6 +24,7 @@ func Images() *schema.Table {
 			transformers.WithSkipFields("Links"),
 		),
 		Relations: []*schema.Table{
+			// ImageMembers(),
```
Comment (dihedron) [inline resources/services/glance/images.go]: Why not attached?

File: resources/services/ironic/allocations.go [hunk 32]
```diff
@@ -0,0 +1,55 @@
...
+	if err != nil {
+		api.Logger().Err(err).Msg("error extracting allocations")
+		return err
+	}
+	for _, allocation := range allAllocations {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		allocation := allocation
```
Comment (dihedron) [inline resources/services/ironic/allocations.go]: Rednundant with Golang 1.22

File: resources/services/ironic/drivers.go [hunk 33]
```diff
@@ -0,0 +1,56 @@
...
+	if err != nil {
+		api.Logger().Err(err).Msg("error extracting drivers")
+		return err
+	}
+	for _, driver := range allDrivers {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		driver := driver
```
Comment (dihedron) [inline resources/services/ironic/drivers.go]: Rednundant with Golang 1.22

File: resources/services/ironic/nodes.go [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+	if err != nil {
+		api.Logger().Err(err).Msg("error extracting nodes")
+		return err
+	}
+	for _, node := range allNodes {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		node := node
```
Comment (dihedron) [inline resources/services/ironic/nodes.go]: Rednundant with Golang 1.22

File: resources/services/ironic/ports.go [hunk 35]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		api.Logger().Err(err).Msg("error extracting ports")
+		return err
+	}
+	for _, port := range allPorts {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		port := port
```
Comment (dihedron) [inline resources/services/ironic/ports.go]: Rednundant with Golang 1.22

File: resources/services/keystone/domains.go [hunk 36]
```diff
@@ -0,0 +1,62 @@
...
+		return err
+	}
+	api.Logger().Debug().Int("count", len(allDomains)).Msg("domains retrieved")
+
+	for _, domain := range allDomains {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		domain := domain
```
Comment (dihedron) [inline resources/services/keystone/domains.go]: Rednundant with Golang 1.22

File: resources/services/keystone/services.go [hunk 37]
```diff
@@ -0,0 +1,68 @@
...
+
+	api := meta.(*client.Client)
+
+	keystone, err := api.GetServiceClient(client.KeystoneV3)
+	if err != nil {
+		api.Logger().Error().Err(err).Msg("error retrieving client")
+		return err
+	}
+
+	service_types := [5]string{"blockstorage", "compute", "identity", "imageservice", "networking"}
```
Comment (dihedron) [inline resources/services/keystone/services.go]: https://docs.openstack.org/keystone/latest/contributor/service-catalog.html

https://pkg.go.dev/github.com/gophercloud/gophercloud/openstack/identity/v3/catalog

HTTP get all'enpoint pubblico di Keystone relativo al catalogo
unmarshal dentro sytriuutura dati
loop su array di catalog entries.

File: resources/services/keystone/services.go [hunk 38]
```diff
@@ -0,0 +1,68 @@
...
+			return err
+		}
+		api.Logger().Debug().Int("count", len(allServices)).Msg("services retrieved")
+
+		for _, service := range allServices {
+			if ctx.Err() != nil {
+				api.Logger().Debug().Msg("context done, exit")
+				break
+			}
+			service := service
```
Comment (dihedron) [inline resources/services/keystone/services.go]: Redundant witrgh go 1.22

File: resources/services/keystone/tenants.go [hunk 39]
```diff
@@ -0,0 +1,62 @@
...
+		return err
+	}
+	api.Logger().Debug().Int("count", len(allTenants)).Msg("users retrieved")
+
+	for _, tenant := range allTenants {
+		if ctx.Err() != nil {
+			api.Logger().Debug().Msg("context done, exit")
+			break
+		}
+		tenant := tenant
```
Comment (dihedron) [inline resources/services/keystone/tenants.go]: Redundant witrgh go 1.22

File: resources/services/nova/images.go [hunk 40]
```diff
@@ -0,0 +1,57 @@
...
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/dihedron/cq-plugin-utils/format"
+	"github.com/dihedron/cq-source-openstack/client"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/images"
+)
+
+func Images() *schema.Table {
```
Comment (dihedron) [inline resources/services/nova/images.go]: Deprecated resource (see Glance Images instead)???

File: resources/services/nova/networks.go [hunk 41]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/dihedron/cq-source-openstack/client"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/networks"
+)
+
+func Networks() *schema.Table {
```
Comment (dihedron) [inline resources/services/nova/networks.go]: Deprecated: remove?!

File: resources/services/nova/secgroups.go [hunk 42]
```diff
@@ -0,0 +1,118 @@
...
+
+	"github.com/apache/arrow/go/v15/arrow"
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/dihedron/cq-plugin-utils/transform"
+	"github.com/dihedron/cq-source-openstack/client"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/secgroups"
+)
+
+func SecGroups() *schema.Table {
```
Comment (dihedron) [inline resources/services/nova/secgroups.go]: Deprecated: Remove?!

File: resources/services/nova/tenantnetworks.go [hunk 43]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"context"
+
+	"github.com/cloudquery/plugin-sdk/v4/schema"
+	"github.com/cloudquery/plugin-sdk/v4/transformers"
+	"github.com/dihedron/cq-source-openstack/client"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/tenantnetworks"
+)
+
+func TenantNetworks() *schema.Table {
```
Comment (dihedron) [inline resources/services/nova/tenantnetworks.go]: Deprecated: renmove?!

File: resources/plugin/client.go (hunk 19 above)
Comment (francescofuggitti) [reply resources/plugin/client.go]: solved in https://github.com/dihedron/cq-plugin-utils/pull/2

File: resources/internal/pattern_matcher/pattern_matcher.go (hunk 20 above)
Comment (francescofuggitti) [reply resources/internal/pattern_matcher/pattern_matcher.go]: solved in https://github.com/dihedron/cq-plugin-utils/pull/2

File: resources/internal/pattern_matcher/pattern_matcher.go (hunk 2 above)
Comment (francescofuggitti) [reply resources/internal/pattern_matcher/pattern_matcher.go]: solved in https://github.com/dihedron/cq-plugin-utils/pull/2

File: resources/internal/pattern_matcher/pattern_matcher.go (hunk 3 above)
Comment (francescofuggitti) [reply resources/internal/pattern_matcher/pattern_matcher.go]: same as above

File: go.mod (hunk 7 above)
Comment (francescofuggitti) [reply go.mod]: done in https://github.com/dihedron/cq-source-openstack/pull/3/commits/e52ad38d17f434f5f440d7bd0f4c5e8be1ad455c

File: go.mod (hunk 8 above)
Comment (francescofuggitti) [reply go.mod]: https://github.com/dihedron/cq-source-openstack/pull/3/commits/e52ad38d17f434f5f440d7bd0f4c5e8be1ad455c

File: resources/internal/utils/data_structs.go (hunk 12 above)
Comment (francescofuggitti) [reply resources/internal/utils/data_structs.go]: Tag is currently used by

- compute/instance_tags
- image/image_tags
- networking/networking_tags

File: plugin/plugin.go (hunk 6 above)
Comment (francescofuggitti) [reply plugin/plugin.go]: It looks like var is needed for ldflags to work [Goreleaser docs](https://goreleaser.com/customization/builds/#building-shared-or-static-libraries)

File: resources/services/cinder/quotasets_usage.go (hunk 26 above)
Comment (francescofuggitti) [reply resources/services/cinder/quotasets_usage.go]: you're right, this was unnecessary

File: resources/services/nova/networks.go (hunk 41 above)
Comment (francescofuggitti) [reply resources/services/nova/networks.go]: removed

File: resources/services/glance/images.go (hunk 31 above)
Comment (francescofuggitti) [reply resources/services/glance/images.go]: only shared images have members. This was solved in https://github.com/dihedron/cq-source-openstack/pull/3/commits/d724a46151eb5ed8b3b251a580c4691cb1bccc8d

File: resources/services/keystone/services.go (hunk 37 above)
Comment (francescofuggitti) [reply resources/services/keystone/services.go]: here the catalog API returns an array of [tokens](https://pkg.go.dev/github.com/gophercloud/gophercloud@v1.11.0/openstack/identity/v3/tokens).[CatalogEntry](https://pkg.go.dev/github.com/gophercloud/gophercloud@v1.11.0/openstack/identity/v3/tokens#CatalogEntry) which is part of the identity V2 (deprecated) APIs. 

For the services resource, we just don't tie the call here to a specific `service_type` but we retrieve all services.

**Pull request merged**